<launch>
    <arg name="bag_file" default="drive_2017-11-18-14-50-05_bosch_big_track_2.bag"/>
    <arg name="bag_image_topic" default="/camera/image_raw"/>
    <arg name="homographied_img_in" value="/camera/warped_image"/>
    <arg name="camera_info" default="/camera/camera_info"/>
    <arg name="homography_in" default="/camera/camera_homography"/>
    <arg name="use_nodelet" default="false"/>
    <arg name="nodelet_manager" default=""/>
    <arg name="run_python" default="false"/>
    <arg name="output" default="screen"/>

    <include file="$(find drive_ros_ximea_importer)/launch/homography_publisher.launch">
        <arg name="vehicle" value="cc2017_car"/>
    </include>

    <include file="$(find drive_ros_image_recognition)/launch/vision.launch">
    </include>

    <include file="$(find drive_ros_localize_visual_odometry)/launch/simple_visual_odometry.launch">
    </include>

    <node pkg="rosbag" type="play" name="player" output="screen" args="-l --clock $(arg bag_file)"/>

    <arg if="$(eval arg('nodelet_manager') == '')" name="manager_used" value="testing_pipeline_nodelet_manager"/>
    <arg unless="$(eval arg('nodelet_manager') == '')" name="manager_used" value="$(arg nodelet_manager)"/>
    <group if="$(arg use_nodelet)">
        <node if="$(eval arg('nodelet_manager') == '')" pkg="nodelet" type="nodelet" name="$(arg manager_used)"  args="manager" output="screen"/>
    </group>

    <include file="$(find drive_ros_image_recognition)/launch/image_processing.launch">
        <arg name="img_in" value="$(arg bag_image_topic)"/>
        <arg name="homographied_img_in" value="$(arg homographied_img_in)"/>
        <arg name="camera_info" value="$(arg camera_info)"/>
        <arg name="homography_in" value="$(arg homography_in)"/>
        <arg name="use_nodelet" value="$(arg use_nodelet)"/>
        <arg name="nodelet_manager" value="$(arg manager_used)"/>
        <arg name="run_python" value="$(arg run_python)"/>
        <rosparam command="load" file="$(find drive_ros_image_recognition)/config/homography.yaml"/>
    </include>
</launch>
